Design/ISIS_Imaging/ISIS_Imaging_general.md

```diff
@@ -109,72 +160,72 @@ Slices - referrs to a reconstructed sinogram. ![Reconstructed slice](https://git
 
-# Finalising name choice
+# 6. Finalising name choice
 
 - MantidImaging
 - ISIS_Imaging
 - Other suggestions?
 
-# Project setup
+# 7. Project setup
 
-## Continous Integration with Coverage
+## 7.1. Continous Integration with Coverage
 
 The package should have a continous integration flow on Pull Requests, which runs all of the CORE and GUI tests. This should also include coverage.
 
 After sphinx documentation is added, the contiguous integration could also check if it builds successfully.
 
-## Installation
+## 7.2. Installation
 
 We can either work towards the package being installable using `python setup.py install`, making it available system wide. The alternative solution is to use it from the folder and provide global access like [the example here](https://mantidproject.github.io/isis_imaging/user_guide/setting_up.html#global-access-to-the-package)
 
-## Testing
+## 7.3. Testing
 
 Testing will use `nose` to run the tests and `coverage` to compute the tests' coverage. The actual unit tests should use the built-in `unittest`, and some might have to use `numpy.testing` for asserting equality for `numpy.ndarrays`.
 
 A module to run the tests has been provided and is called `run_tests.py`, in the root of the repository. It takes as an argument the names of different packages, and then runs the tests specifically for them. The possible arguments are listed via `-h` flag, or at the top of the file in the source.
 
 There is currently no integration with Mantid's testing.
 
-### Core
+### 7.3.1. Core
 
-Every module should have associated unit tests, unless there is good reason not to have one. 
+Every module should have associated unit tests, unless there is good reason not to have one.
 
 There should also be a collection of system tests to make sure that the functionality works on a higher level than unit tests.
 
-### GUI
+### 7.3.2. GUI
 
-The GUI should use the MVP pattern, making it easier to mock and unit test. Mocking should be done with the python built-in `mock`. There should be an associated mock and presenter unit testing for every MVP used. 
+The GUI should use the MVP pattern, making it easier to mock and unit test. Mocking should be done with the python built-in `mock`. There should be an associated mock and presenter unit testing for every MVP used.
 
 Simpler cases like the `load`/`save` dialogues do not need to use MVP, because they have very little, if any, logic.
 
-# Guidelines for ISIS Imaging CORE
+# 8. Guidelines for ISIS Imaging CORE
 
-## CUDA filter expansion
+## 8.1. CUDA filter expansion
 
 Integrating a package that supports CUDA has potential for improvement and expansion of the imaging filters.
 
 A potential problem for CUDA is for machines with low VRAM, transferring the data multiple times might be slower that just doing it in a bulk in the CPU, if it has a lot of RAM.
```

Comment:
Is there a way to query the amount of VRAM on a system and make a decision based on the result?